pkg/liquidity-mining/contracts/gauges/StakelessGaugeController.sol

```diff
@@ -0,0 +1,223 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IAuthorizerAdaptor.sol";
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IGaugeAdder.sol";
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IGaugeController.sol";
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IStakelessGauge.sol";
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IStakelessGaugeController.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/Address.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/EnumerableSet.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/ReentrancyGuard.sol";
+
+import "./arbitrum/ArbitrumRootGauge.sol";
+
+/**
+ * @title Stakeless Gauge Controller
+ * @notice Implements IStakelessGaugeController; refer to it for API documentation.
+ */
+contract StakelessGaugeController is IStakelessGaugeController, ReentrancyGuard {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    mapping(IGaugeAdder.GaugeType => EnumerableSet.AddressSet) private _gauges;
+    IAuthorizerAdaptor private immutable _authorizerAdaptor;
+    IGaugeController private immutable _gaugeController;
+    IGaugeAdder private immutable _gaugeAdder;
+
+    constructor(
+        IAuthorizerAdaptor authorizerAdaptor,
+        IGaugeController gaugeController,
+        IGaugeAdder gaugeAdder
+    ) {
+        _authorizerAdaptor = authorizerAdaptor;
+        _gaugeController = gaugeController;
+        _gaugeAdder = gaugeAdder;
+    }
+
+    modifier gaugeTypeChecked(IGaugeAdder.GaugeType gaugeType) {
+        require(_isValidGaugeType(gaugeType), "Unsupported gauge type");
```

Comment:
If you're calling this concept 'valid gauge type', then I'd call the modifier `withValidGaugeType(type)` instead. Similar to the comment in add/register, try to avoid introducing new terms ('type checked' vs 'valid type').

Reply:
+1; done.